packages/plugins/typescript/graphql-request/src/visitor.ts

```diff
@@ -116,20 +116,32 @@ export class GraphQLRequestVisitor extends ClientSideBaseVisitor<
           }
           return `${operationName}(variables${optionalVariables ? '?' : ''}: ${
             o.operationVariablesTypes
-          }, requestHeaders?: Dom.RequestInit["headers"]): Promise<{ data: ${o.operationResultType}; extensions?: ${
-            this.config.extensionsType
-          }; headers: Dom.Headers; status: number; }> {
-    return withWrapper((wrappedRequestHeaders) => client.rawRequest<${
-      o.operationResultType
-    }>(${docArg}, variables, {...requestHeaders, ...wrappedRequestHeaders}), '${operationName}', '${operationType}');
+          }, signal?: AbortSignal, requestHeaders?: Dom.RequestInit["headers"]): Promise<{ data: ${
```

Comment:
What about accepting `RequestInit` in general instead of adding each parameter one by one